### DISCUSSION TITLE
Error: [app/unknown] Configuration fails. It may be caused by an invalid GOOGLE_APP_ID in GoogleService-Info.plist or set in the customized options.

### DISCUSSION THREAD

**Discussion by pandam77** [invertase/react-native-firebase#7271] [Q&A]
I get an error when I use initializeApp().
My firebase config settings are correct, but why?

> WARN  Possible Unhandled Promise Rejection (id: 0):
> Error: [app/unknown] Configuration fails. It may be caused by an invalid GOOGLE_APP_ID in GoogleService-Info.plist or set in the customized options.
> NativeFirebaseError: [app/unknown] Configuration fails. It may be caused by an invalid GOOGLE_APP_ID in GoogleService-Info.plist or set in the customized options.
>     at initializeApp (http://192.168.2.51:8081/node_modules%5Cexpo%5CAppEntry.bundle//&platform=ios&dev=true&hot=false:132742:59)
>     at ?anon_0_ (http://192.168.2.51:8081/node_modules%5Cexpo%5CAppEntry.bundle//&platform=ios&dev=true&hot=false:131330:45)
>     at next (native)
>     at asyncGeneratorStep (http://192.168.2.51:8081/node_modules%5Cexpo%5CAppEntry.bundle//&platform=ios&dev=true&hot=false:4644:26)
>     at _next (http://192.168.2.51:8081/node_modules%5Cexpo%5CAppEntry.bundle//&platform=ios&dev=true&hot=false:4663:29)
>     at anonymous (http://192.168.2.51:8081/node_modules%5Cexpo%5CAppEntry.bundle//&platform=ios&dev=true&hot=false:4668:14)
>     at tryCallTwo (/Users/distiller/react-native/sdks/hermes/build_iphoneos/lib/InternalBytecode/InternalBytecode.js:61:9)
>     at doResolve (/Users/distiller/react-native/sdks/hermes/build_iphoneos/lib/InternalBytecode/InternalBytecode.js:216:25)
>     at Promise (/Users/distiller/react-native/sdks/hermes/build_iphoneos/lib/InternalBytecode/InternalBytecode.js:82:14)
>     at anonymous (http://192.168.2.51:8081/node_modules%5Cexpo%5CAppEntry.bundle//&platform=ios&dev=true&hot=false:4660:25)
>     at apply (native)

"expo": "~48.0.18",
"@react-native-firebase/analytics": "^18.1.0",
"@react-native-firebase/app": "^18.1.0",

firebase.tsx
> import firebase from "@react-native-firebase/app";
> import analytics from "@react-native-firebase/analytics";
> const firebaseConfig = {
>    your firebase config
> };
> const firebase_config_name = { name: 'SECONDARY_APP', };
> await firebase.initializeApp(firebaseConfig, firebase_config_name);
> analytics().logEvent("test_analytics_event", {
>     additionaParam: "test",
> });

app.json
> "plugins": [
>       "@react-native-firebase/app",
>       [
>         "expo-build-properties",
>         {
>           "ios": {
>             "useFrameworks": "static"
>           }
>         }
>       ]
>     ]

**Comment by hidayat-nur** [comment]
same with me, and don't know to solve this : (

**Comment by Artem-Pavlenko** [comment]
@pandam77 have you found a solution to that?

**Comment by AndresHB** [comment]
It's weird, after updating to RN v0.72.6 I started getting this error, even without updating the Firebase dependencies versions.
Before upgrading to the RN version, I didn't have to call firebase.initializeApp because my app used GoogleService-Info.plist and google-services.json natively.

**Comment by gavrielk** [comment]
Same issue here with RN 0.72.4

**Comment by mikehardy** [comment]
I recognize multiple people are reporting this but it doesn't rise to the level of an issue because it isn't detailed or reproducible
https://stackoverflow.com/help/how-to-ask
https://stackoverflow.com/help/minimal-reproducible-example

Note that it is possible to create firebase projects that are on the free tier so will not incur billing, and in that way create reproducible examples

Also note that we test primary + secondary app configuration in our e2e tests all the time and it works. There is some hint on an unrelated issue that if your projects are part of a completely different organization you may not be able to initialize them as a secondary app

There is also a strict requirement that you *do* initialize the primary app via google-services.json/GoogleServices-Info.plist - that is not something you can avoid and you should download fresh copies of each to make sure that is working. If you don't have secondary apps there is nothing further to do after downloading fresh copies + re-running pod install / expo prebuild --> it should work

**Comment by gavrielk** [comment]
I used a single app and still got this issue so just want to share what finally fixed the issue for me.
 I've been using the react firebase library in the past which worked for a while until it stopped working and I switched to react-native-firebase.
Assuming you have the google service files in place for both ios and android.
**For Android:** I removed the call for firebase.initializeApp, and it started working fine from there, apparently it is stated [here](https://v5.rnfirebase.io/docs/master/core/default-app).
**For iOS:** Removing firebase.initializeApp caused other issues:
 ERROR  [Error: No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp()]
, so it had be stay but I had typo in the config as I used web instead of ios in the appId, so be careful...
